mantidproject/mantid
# Reflectometry GUI restructuring 3: missing widgets

### PRs ~~#23630~~ and ~~#24345~~ should be merged first
This can be reviewed in the meantime by comparing against [23051_restructure_refl_save_tab](https://github.com/gemmaguest/mantid/tree/23051_restructure_refl_save_tab)

**Description of work.**
Ongoing work to integrate the `JobTreeView` into the `ISIS Reflectometry` GUI. This PR is being merged to a feature branch because not all functionality is implemented yet.

- Include GUI widgets that were recently added on master to the restructured files. The changes were not automatically merged in the original restructuring in PR #23630 because the files have moved.
- MVP models have been created for the new GUI components in line with the restructured design.
- Fix an issue with the equality operator for polarization corrections not checking the type

**Report to:** nobody

**To test:**

- Open the `ISIS Reflectometry` GUI
- Click the `Experiment Settings` tab
- Check that the `Debug` check box is enabled. It should be unticked by default.
- Check that the `IncludePartialBins` check box is only enabled when `SummationType` is `SumInQ`. It should be unticked by default.
- Check that the `FloodCorrection` combo box contains the options `Workspace` and `ParameterFile`. The `Flood Workspace` combo box should only be visible when the `FloodCorrection` type is set to `Workspace`.
- Load some workspaces and check that they become available in the `Flood Workspace` combo box. Note that table and group workspaces should be excluded.
- Note that processing is currently disabled on the GUI so although the models are updated with the contents of the GUI they are not actually used. Please just review the code.

Refs #23043

*This does not require release notes* because it only includes under-the-hood changes.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
